File: 2025/2.ts
import { run } from "@util";

async function part1(input: string): Promise<number> {
  const pattern = /^(.+)\1$/;

  let s = 0;

  for (const line of input.split(",")) {
    const [a, b] = line.split("-").map(Number);

    for (let id = a; id <= b; id++) {
      if (id.toString().match(pattern)) {
        s += id;
      }
    }
  }

  return s;
}

async function part2(input: string): Promise<number> {
  const pattern = /^(.+)\1+$/;

  let s = 0;

  for (const line of input.split(",")) {
    const [a, b] = line.split("-").map(Number);

    for (let id = a; id <= b; id++) {
      if (id.toString().match(pattern)) {
        s += id;
      }
    }
  }

  return s;
}

run(2, 2025, part1, part2);

File: 2025/2.py
import re

import util


def part1(input: str) -> int:
    pattern = re.compile(r"^(.+)\1$")

    s = 0
    for r in input.split(","):
        a, b = map(int, r.split("-"))
        for i in range(a, b + 1):
            if pattern.match(str(i)) is not None:
                s += i

    return s


def part2(input: str) -> int:
    pattern = re.compile(r"^(.+)\1+$")

    s = 0
    for r in input.split(","):
        a, b = map(int, r.split("-"))
        for i in range(a, b + 1):
            if pattern.match(str(i)) is not None:
                s += i
    return s


if __name__ == "__main__":
    util.run(2, part1, part2)
